src/hooks/useCart.ts

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect, useCallback } from "react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../modules/Store";
+import { getCarts, deleteCart, totalCheck } from "../modules/Carts";
+import { postOrder } from "../modules/Orders";
+
+function useCart() {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const [totalPrice, setTotalPrice] = useState(0);
+  const [totalChecked, setTotalChecked] = useState(true);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+
+  const carts = useSelector((state: RootState) => state.carts.cartsList);
+
+  const checkedCartList = carts
+    .filter((cart) => cart.checked === true)
+    .map((cart) => {
+      return { ...cart.product, quantity: cart.quantity };
+    });
```

Comment:
메서드 체이닝이 깔끔하게 되어 있어 가독성이 좋네요!

```suggestion
  const checkedCartList = carts
    .filter((cart) => cart.checked)
    .map((cart) => {
      return { ...cart.product, quantity: cart.quantity };
    });
```

이런 식으로 축약해서 사용해보시는건 어떨까요?